src/test/java/org/kiwiproject/dropwizard/util/health/MonitoredJobHealthCheckTest.java

```diff
@@ -0,0 +1,270 @@
+                        .expectedFrequency(Duration.seconds(1))
+                        .build();
+            }
+
+            @Test
+            void becauseJobHasNeverFailed() {
+                when(job.getLastSuccess()).thenReturn(new AtomicLong(now));
+                when(job.isActive()).thenReturn(true);
+                when(job.getLastFailure()).thenReturn(new AtomicLong());
+
+                assertHealthyHealthCheck(healthCheck, "Last successful execution was: {}", Instant.ofEpochMilli(now).toString());
+            }
+
+            @Test
+            void becauseJobFailedOutsideOfThreshold() {
+                when(job.getLastSuccess()).thenReturn(new AtomicLong(now));
+                when(job.isActive()).thenReturn(true);
+
+                var failureOutsideThreshold = now - MonitoredJobHealthCheck.DEFAULT_WARNING_DURATION.toMilliseconds() - 1;
+                when(job.getLastFailure()).thenReturn(new AtomicLong(failureOutsideThreshold));
+
+                assertHealthyHealthCheck(healthCheck, "Last successful execution was: {}", Instant.ofEpochMilli(now).toString());
+            }
+        }
+
+        @Nested
+        class WhenLastRunDoesNotMeetThreshold {
+
+            @Test
+            void becauseLastRunIsCloseEnough() {
+                when(job.getLastSuccess()).thenReturn(new AtomicLong(now));
+                when(job.isActive()).thenReturn(true);
+                when(job.getLastFailure()).thenReturn(new AtomicLong());
+
+                var env = mock(KiwiEnvironment.class);
+                when(env.currentTimeMillis())
+                        .thenReturn(now - MonitoredJobHealthCheck.MINIMUM_WARNING_THRESHOLD.toMilliseconds() - 1)
+                        .thenReturn(now);
+
+                var healthCheck = MonitoredJobHealthCheck.builder()
+                        .job(job)
+                        .expectedFrequency(Duration.seconds(1))
+                        .environment(env)
+                        .build();
+
+                assertHealthyHealthCheck(healthCheck, "Last successful execution was: {}", Instant.ofEpochMilli(now).toString());
+            }
+
+            @Test
+            void becauseCheckRanRightAfterStartup_BeforeFirstRun() {
+                when(job.getLastSuccess()).thenReturn(new AtomicLong());
+                when(job.isActive()).thenReturn(true);
+                when(job.getLastFailure()).thenReturn(new AtomicLong());
+
+                var env = mock(KiwiEnvironment.class);
+                when(env.currentTimeMillis())
+                        .thenReturn(now)
+                        .thenReturn(now);
+
+                var healthCheck = MonitoredJobHealthCheck.builder()
+                        .job(job)
+                        .expectedFrequency(Duration.seconds(1))
+                        .environment(env)
+                        .build();
+
+                assertHealthyHealthCheck(healthCheck, "Last successful execution was: never");
```

Comment:
I guess it'll only happen once, but if the first execution of the health check succeeds, then this message is a bit odd. But not sure how to handle without special handling for the case when the check has never run previously...

Reply:
Leaving it for now as this is what we have today.  We can re-look at it later.